integration/class-enable-mastodon-apps.php

```diff
@@ -26,14 +29,84 @@ class Enable_Mastodon_Apps {
 	 * Initialize the class, registering WordPress hooks
 	 */
 	public static function init() {
+		\add_filter( 'mastodon_api_mapback_user_id', array( self::class, 'maybe_map_user_to_blog' ) );
 		\add_filter( 'mastodon_api_account_followers', array( self::class, 'api_account_followers' ), 10, 2 );
 		\add_filter( 'mastodon_api_account', array( self::class, 'api_account_add_followers' ), 20, 2 );
 		\add_filter( 'mastodon_api_account', array( self::class, 'api_account_external' ), 15, 2 );
+		\add_filter( 'mastodon_api_account', array( self::class, 'api_account_internal' ), 9, 2 );
 		\add_filter( 'mastodon_api_search', array( self::class, 'api_search' ), 40, 2 );
 		\add_filter( 'mastodon_api_search', array( self::class, 'api_search_by_url' ), 40, 2 );
 		\add_filter( 'mastodon_api_get_posts_query_args', array( self::class, 'api_get_posts_query_args' ) );
 		\add_filter( 'mastodon_api_statuses', array( self::class, 'api_statuses_external' ), 10, 2 );
 		\add_filter( 'mastodon_api_status_context', array( self::class, 'api_get_replies' ), 10, 23 );
+		\add_action( 'mastodon_api_update_credentials', array( self::class, 'api_update_credentials' ), 10, 2 );
+	}
+
+	/**
+	 * Map user to blog if user is disabled
+	 *
+	 * @param int $user_id The user id
+	 *
+	 * @return int The user id
+	 */
+	public static function maybe_map_user_to_blog( $user_id ) {
```

Comment:
is this working?

Reply:
It is! I use it inside a couple of our callbacks to change somebody's WP User ID to the Blog User ID only when the user type is disabled and the blog is enabled. In this way a user who is logged in is doing what the plugin supports: only the Blog User. It does not allow us to log in as the Blog User when both Blog and User types are enabled, but that seems to be a much smaller problem overall.

<img width="316" alt="Screenshot 2024-06-26 at 12 48 08" src="https://github.com/Automattic/wordpress-activitypub/assets/195089/66102f3f-fe90-4804-8de8-f424371effbf">